static/third/jquery-filedrop/jquery.filedrop.js

```diff
@@ -362,12 +362,7 @@
 
         // Pass any errors to the error option
         if (xhr.status < 200 || xhr.status > 299) {
-          if (this.responseText.includes("Upload would exceed your maximum quota.")) {
```

Comment:
Done and updated the commit comment to explain the reasoning behind choosing erverResponse.msg over xhr.statusText.

Reply:
Got it, thanks. I just read through some of this code (in the filedrop library and in our code using it), and I see what's going on.

This argument becomes the first argument to a call to the filedrop call's `error` callback (search for `on_error` in the filedrop file), which for us is `uploadError` in `upload.js`.

All the other calls to that callback pass a machine-readable code string as that first argument -- like `BrowserNotSupported`. I wouldn't want to mix sometimes sending machine-readable codes, sometimes machine-opaque text, through the same channel -- that's what leads to broken logic like what you caught here, where code tries to inspect the user-facing text that should be opaque to it. So let's keep this first argument machine-readable.

Fortunately, there are a couple of things we can do to make that work nicely!
* We have a mechanism for the backend to send the frontend a nice machine-readable code, alongside the user-facing message: just pass a second argument to the `JsonableError` constructor, with a value like `ErrorCode.QUOTA_EXCEEDED`.
  * Or an alternate form: make a simple JsonableError subclass. It would look a lot like the `ExceededQuotaError` class you deleted earlier. :-p (Which we apparently never actually used. Also, `QuotaExceededError` would be a better name.)
  * Either way, we'd then have `serverResponse.code` equal to the string `'QUOTA_EXCEEDED'`.
* As it turns out, the only thing that uses that `on_error` feature is that one place at line 534 of `jquery.filedrop.js`; and the one thing that uses `filedrop` at all is that one place in `upload.js`. So we have total control over this interface, and can make it better.
  * Let's replace the "status" parameter, up and down the whole chain, with "message". Then we can pass `serverResponse.msg` there, and in `uploadError` we can pass the message straight through.

Reply:
Also, a further refinement:
* Let's make `RealmUploadLimitException` just be that `JsonableError` subclass; then there's no need to transmute it later into a different exception with a try/except and `json_error`.
  * Let's call it `RealmUploadQuotaError`, and use a code named `REALM_UPLOAD_QUOTA`.
  * The message can be passed to the `RealmUploadQuotaError` constructor, just like `JsonableError`.